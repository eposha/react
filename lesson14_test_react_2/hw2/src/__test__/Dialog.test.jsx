import React from "react";
import { shallow } from "enzyme";
import Dialog from "../Dialog.jsx";

describe("Dialog", () => {
  const mockCallback = jest.fn();
  it("should be display component Dialog if prop isOpen = true", () => {
    const wrappedComponent = shallow(<Dialog isOpen onClose={mockCallback} />);
    expect(wrappedComponent.find(".dialog").exists()).toBeTruthy();
  });

  it("shouldn`t display component Dialog if prop isOpen = false", () => {
    const wrappedComponent = shallow(<Dialog onClose={mockCallback} />);
    expect(wrappedComponent.find(".dialog").exists()).toEqual(false);
  });

  it("should display text from prop title", () => {
    const wrappedComponent = shallow(
      <Dialog isOpen title={"text"} onClose={mockCallback} />
    );
    expect(wrappedComponent.find("h4").text()).toEqual("text");
  });

  it("should display data from prop children", () => {
    const wrappedComponent = shallow(
      <Dialog isOpen children={"text"} onClose={mockCallback} />
    );
    expect(wrappedComponent.find(".dialog__content").text()).toEqual("text");
  });

  it("component Dialog should display children elements", () => {
    const wrappedComponent = shallow(<Dialog isOpen onClose={mockCallback} />);
    expect(wrappedComponent).toMatchSnapshot();
  });
});
