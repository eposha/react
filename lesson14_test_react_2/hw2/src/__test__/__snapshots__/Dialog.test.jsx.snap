// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component Dialog should display children elements 1`] = `
<div
  className="dialog"
>
  <div
    className="dialog__heading"
  >
    <h4
      className="dialog__title"
    />
    <button
      className="dialog__close-btn"
      onClick={[MockFunction]}
    >
      +
    </button>
  </div>
  <div
    className="dialog__content"
  />
</div>
`;
